packages/dds/tree/src/feature-libraries/indexing/anchorTreeIndex.ts

```diff
@@ -82,6 +82,8 @@ export class AnchorTreeIndex<TKey extends TreeIndexKey, TValue>
 	 * @param getValue - a pure and functional function that returns the associated value of one or more anchor nodes, can be used to map and filter the indexed anchor nodes
 	 * so that the values returned from the index are more usable
 	 * @param checkTreeStatus - a function that gets the tree status from an anchor node, used for filtering out detached nodes
+	 * @param reIndexSpine - as a performance optimization, re-indexing up the spine can be turned off for indexes that do not need it, this only applies to indexes that only allow
```

Comment:
slightly confusing wording. I think you're saying that reIndexSpine should only ever be true if you have this shallow style indexing scheme, right? if that's the only time you can ever pass true and be correct, maybe rename the flag to isShallowIndex or something and document that it optimizes this case